index.js

```diff
@@ -112,6 +122,43 @@ class Conf {
 		}
 	}
 
+	_migrate(migrations, versionToMigrate = '0.0.0') {
+		const MIGRATION_KEY = '__conf-migrated-version__';
+
+		const previousMigratedVersion = this.get(MIGRATION_KEY, '0.0.0');
+
+		const newerVersions = Object.keys(migrations)
+			.filter(version => this._shouldPerformMigration(version, previousMigratedVersion, versionToMigrate))
+			.sort(semver.compare);
+
+		const migrationsToRun = newerVersions.map(version => migrations[version]);
+
+		migrationsToRun.forEach(migration => migration(this));
+
+		this.set(MIGRATION_KEY, versionToMigrate);
+	}
+
+	/**
+	 * Specifies if a version is suitable to run its migrations.
+	 * To be suitable, a version should be comprehended between the previous migrated version and the version desired to migrate.
+	 *
+	 * @param {string} version The version that's being tested to be suitable or not.
+	 * @param {string} previousMigratedVersion The version matching the last migration that ocurred.
+	 * @param {string} versionToMigrate The version that specifies what to migrate to (a.k.a the current version).
+	 * @returns {boolean} If the version is suitable or not.
+	 */
+	_shouldPerformMigration(version, previousMigratedVersion, versionToMigrate) {
```

Comment:
Just a small suggestion, what if we rename `version` to `canditateVersion` or something similar to make it more obvious?

Reply:
Sure 👍